qemu/tests/x86_cpu_flag_la57.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from virttest import cpu
+from virttest import utils_misc
+from virttest import env_process
+from virttest import error_context
+
+from provider.cpu_utils import check_cpu_flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu flag la57.
+    1) Check if current flags are in the supported lists on host, if no, cancel test
+    2) Otherwise, boot guest without the flag and without Icelake-Server model
+    3) Check cpu flags inside guest,shouldn't be there(only for linux guest)
+    4) Check the virtual address inside guest, should be '48 bits'
+    5) Boot guest with the flag
+    6) Check cpu flags inside guest,should be there(only for linux guest)
+    7) Check the virtual address inside guest, should be '57 bits'
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    flags = params["flags"]
+    check_host_flags = params.get_boolean("check_host_flags")
+    if check_host_flags:
+        check_cpu_flags(params, flags, test)
+
+    host_cpu_models = cpu.get_host_cpu_models()
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    qemu_cpu_models = cpu.get_qemu_cpu_models(qemu_binary)
+    for cpu_model in host_cpu_models:
+        if cpu_model != 'Icelake-Server':
+            if cpu_model in qemu_cpu_models:
+                params["cpu_model"] = cpu_model
+                break
+
+    for i in range(2):
+        flags = ''
+        check_cmd = params['check_cmd']
+        if i >= 1:
+            params['cpu_model_flags'] += ",+la57"
+            flags = 'la57'
+            params['no_flags'] = ''
+            check_cmd = params['check_cmd_la57']
```

Comment:
Better drop this for loop and separate them into different variants if you have so many different params to define:
```
    variants:
        - with_la57:
            cpu_model_flags += ",+la57"
            flags = 'la57'
            'no_flags' = ''
            bits_virtual = '57'
        - without_la57:
            bits_virtual = '48'
    check_cmd = 'grep "address sizes" /proc/cpuinfo | uniq | grep "${bits_virtual} bits virtual"'
```

Reply:
@vivianQizhu Thanks and update.